sunkit_image/coalignment.py

```diff
@@ -1,284 +1,55 @@
-    """
-    Calculate an estimate of the location of the peak of the correlation result
-    in image pixels.
-
-    Parameters
-    ----------
-    corr : `numpy.ndarray`
-        A 2D correlation array.
-
-    Returns
-    -------
-    `~astropy.units.Quantity`
-        The shift amounts ``(y, x)`` in image pixels. Subpixel values are
-        possible.
-    """
-    # Get the index of the maximum in the correlation function
-    ij = np.unravel_index(np.argmax(corr), corr.shape)
-    cor_max_x, cor_max_y = ij[::-1]
-
-    # Get the correlation function around the maximum
-    array_maximum = corr[
-        np.max([0, cor_max_y - 1]) : np.min([cor_max_y + 2, corr.shape[0] - 1]),
-        np.max([0, cor_max_x - 1]) : np.min([cor_max_x + 2, corr.shape[1] - 1]),
-    ]
-    y_shift_maximum, x_shift_maximum = _get_correlation_shifts(array_maximum)
 
-    # Get shift relative to correlation array
-    y_shift_correlation_array = y_shift_maximum + cor_max_y * u.pix
-    x_shift_correlation_array = x_shift_maximum + cor_max_x * u.pix
+def register_coalignment_method(name, method):
+    registered_methods[name] = method
 
-    return y_shift_correlation_array, x_shift_correlation_array
 
+############################ Coalignment Interface begins #################################
+def replace_nan_with_mean(array):
+    mean_value = np.nanmean(array)
+    array[np.isnan(array)] = mean_value
 
-def _get_correlation_shifts(array):
-    """
-    Estimate the location of the maximum of a fit to the input array. The
-    estimation in the "x" and "y" directions are done separately. The location
-    estimates can be used to implement subpixel shifts between two different
-    images.
-
-    Parameters
-    ----------
-    array : `numpy.ndarray`
-        An array with at least one dimension that has three elements. The
-        input array is at most a 3x3 array of correlation values calculated
-        by matching a template to an image.
-
-    Returns
-    -------
-    `~astropy.units.Quantity`
-        The ``(y, x)`` location of the peak of a parabolic fit, in image pixels.
-    """
-    # Check input shape
-    ny = array.shape[0]
-    nx = array.shape[1]
-    if nx > 3 or ny > 3:
-        msg = "Input array dimension should not be greater than 3 in any dimension."
-        raise ValueError(msg)
-
-    # Find where the maximum of the input array is
-    ij = np.unravel_index(np.argmax(array), array.shape)
-    x_max_location, y_max_location = ij[::-1]
-
-    # Estimate the location of the parabolic peak if there is enough data.
-    # Otherwise, just return the location of the maximum in a particular
-    # direction.
-    y_location = _parabolic_turning_point(array[:, x_max_location]) if ny == 3 else 1.0 * y_max_location
+    return array
 
-    x_location = _parabolic_turning_point(array[y_max_location, :]) if nx == 3 else 1.0 * x_max_location
 
-    return y_location * u.pix, x_location * u.pix
+def convert_map_to_array(map_obj):
```

Comment:
Why is this needed?

Reply:
So Will and I had discussed that while building coalignment from ground up, we would be extracting data array and dispatching the numpy arrays to the method. Although this function, in particular might seem redundant, as it can be executed in a single line.

Reply:
That makes sense but if you have a map you can always just access the array directly, would that not be a more ideal way or is there a need to do it like this?

Reply:
Yeah that makes sense, we will need to iterate on the design as we make changes.